### PR TITLE
Resolve bugs 1

### PR DESCRIPTION
**Pull request by artemkhere**
Fix a series of bugs that I was able to identify.

### CODE REVIEW

File: Guides/RepeatWords/Playing.js [hunk 1]
```diff
@@ -8,14 +8,12 @@ export default class Playing extends Component {
...
             wordsOpacity: new Animated.Value(0),
-            progressValueListener: null,
-            progressCircleAnimationProcess: null,
         }
     }
 
     componentDidMount() { this.startInstructionsTextAnimation(); }
 
-    componentWillUnmount() { Animated.timing(this.state.wordsOpacity).stop(); }
+    componentWillUnmount() { Animated.sequence(this.state.wordsOpacity).stop(); }
```
Comment (peteygao) [inline Guides/RepeatWords/Playing.js:16]: What’s the difference between `timing` and `sequence`?

File: SessionPlayer/SessionPlayer.js [hunk 2]
```diff
@@ -137,14 +137,14 @@ export default class SessionPlayer extends Component {
...
-        const { guideState, session } = this.state;
+        const { guideState, currentGuideIndex, session } = this.state;
 
         switch(guideState) {
             case 'playing':
                 this.setState({
                     guideState: 'paused',
                     currentInstructionIndex: 0,
-                    currentInstruction: session[0].instructions[0],
+                    currentInstruction: session[currentGuideIndex].instructions[0],
```
Comment (peteygao) [inline SessionPlayer/SessionPlayer.js:147]: Oh, okay, this is a different behaviour than I was expecting, but I understand.

**Review by peteygao — APPROVED** [review]
LGTM!

File: Guides/RepeatWords/Playing.js (hunk 1 above)
Comment (artemkhere) [reply Guides/RepeatWords/Playing.js:16]: `sequence` takes an array of `timing` animations. This `stop()` actually stops all of the animations. Before it didn't.

**Pull request merged**